### PR TITLE
work on ImgPostControllerIntTest

### LINKED ISSUE #35: [Testing] ImgPost Controller Test
### Resources: 
Resource from Azhya: 
[Spring Boot Testing Basics: How to Unit Test & Integration Test REST Controllers](https://www.youtube.com/watch?v=pNiRNRgi5Ws)

Additional Resources:
[Test Driven Development with Spring Boot - Sannidhi Jalukar, Madhura Bhave](https://www.youtube.com/watch?v=s9vt6UJiHg4)

-----
### Methods: 
As a small reminder, we only need the minimum amount of controller code needed to actually run the test (Calvin's example should show that). 
Note: It's expected to fail. 
Action item: Please go ahead and set up the minimum amount of code (in the Controller layer) for your assigned testing table that you will need to create your test. 

Your tests should have the following methods: 
- getAll
- getById
- Create
- Update
- Delete

### PR DESCRIPTION
**Pull request by pyaeger**
Five tests failed accordingly. Need more time to implement creation and modification tests.

I had mistakenly worked on the ImgPost model. Submitting that in case that's helpful as a starting point for whoever is assigned.

### CODE REVIEW

**Comment by pyaeger** [comment]
I don't know why there were changes in the .classpath file.

**Review by FullSol — DISMISSED** [review]
Looks good. And looks like you wrote out all of the code for the model. Easy mode for whoever gets assigned to that! I think I did like two of them now...

**Comment by FullSol** [comment]
What happened here? Looks like User model was pasted on top of the ImgPost model?

File: .classpath
Comment (FullSol) [inline .classpath]: Was this removed? I thought I approved the changes that were here before?

**Comment by pyaeger** [comment]
GitHub showed there was a conflict (overlapping redundant sectiosn) between my to-be-merged branch ImgPost model and dev ImgPost model. I went in through the web editor here in GitHub as suggested by GitHub and resolved the conflict by removing my sections.

**Comment by FullSol** [comment]
> 

Is there an undo function? Whatever was there before I approved.

**Comment by pyaeger** [comment]
When I worked on the ImgPost model it was empty. Looks like GitHub now sees the model now in dev and saw that the ImgPost model I submitted in my pull request was overlapping/conflicting with the dev ImgPost. It asked that I resolve the conflict in order for the pull request to proceed.

**Comment by FullSol** [comment]
> When I worked on the ImgPost model it was empty. Looks like GitHub now sees the model now in dev and saw that the ImgPost model I submitted in my pull request was overlapping/conflicting with the dev ImgPost. It asked that I resolve the conflict in order for the pull request to proceed.

I see what is happening. I thought you made the changes to Impost, but that was me last night... But when there are merging differences the code needs to be reviewed and the more accurate code needs to be accepted.

But this is why we have reviews, so not a big deal

Just an FYI the ImgPost model would be obliterated when this branch was merged. I'm not sure how to undo the merge you just made, but I'm sure we can figure it out.

**Comment by FullSol** [comment]
I will say though, in the future, let's not edit code once an approval has been made unless changes are requested. Doing so invalidates the approver's review and if it is not seen fast enough we may run into an issue.

**Comment by pyaeger** [comment]
Okay sounds good. Will do going forward. 

(I don't think the merge will obliterate your ImgPost model in dev but rather **_merge_** the few sections that remain in my proposed pull request ImgPost model into yours now that there are no conflicts/duplicate lines.)

**Comment by FullSol** [comment]
> (I don't think the merge will obliterate your ImgPost model in dev but rather **_merge_** the few sections that remain in my proposed pull request ImgPost model into yours now that there are no conflicts/duplicate lines.)

No, that is not correct in this case. If you merge a pull request with  newer information and the old code is not present it will overwrite the branch you are merging into. That error you were getting from GitHub was the chance to do what you are talking about and choosing what code to keep and what code to remove.

I've reverted that change and fixed any errors so the code should be fine to merge not, If not it's my fault and I'll make a ticket to fix it later.

**Comment by pyaeger** [comment]
Okay, sorry for problem there. Ugh.

**Comment by FullSol** [comment]
Not a problem at all. Learning experience! Sort of the point of all of this.

**Review by FullSol — APPROVED** [review]

**Comment by pyaeger** [comment]
Is this still in the queue for merging pending the final review? Thanks for the clarification. :)

File: src/test/java/com/amplifier/controllers/ImgPostControllerIntTest.java
Comment (L3viathon) [inline src/test/java/com/amplifier/controllers/ImgPostControllerIntTest.java]: For "`mockImgPost1 = new Impost(1, "ImageTest");`"?, "`imageTest`" needs to be a mockUser (aka author). You will have to create a mock object of user to add when initializing `mockImgPost1`. I've added the ImgPost constructor for your reference below. 

```
 public ImgPost(String imgLocation, User author) {
        this.imgLocation = imgLocation;
        this.author = author;
    }
```

File: src/test/java/com/amplifier/controllers/ImgPostControllerIntTest.java
Comment (L3viathon) [inline src/test/java/com/amplifier/controllers/ImgPostControllerIntTest.java]: "`id=xxxxxxxxxxxxxxxxxxxx`" should be 1 (for ex). This id is referring to imgPost id.

File: src/test/java/com/amplifier/controllers/ImgPostControllerIntTest.java
Comment (L3viathon) [inline src/test/java/com/amplifier/controllers/ImgPostControllerIntTest.java]: "`id=xxxxxxxxxxxxxxxxxxxx`" should be 1 (for ex). This id is referring to imgPost id.

File: src/test/java/com/amplifier/controllers/ImgPostControllerIntTest.java
Comment (L3viathon) [inline src/test/java/com/amplifier/controllers/ImgPostControllerIntTest.java]: "`id=xxxxxxxxxxxxxxxxxxxx`" should be 1 (for ex). This id is referring to imgPost id.

**Review by L3viathon — CHANGES_REQUESTED** [review]

File: src/test/java/com/amplifier/controllers/ImgPostControllerIntTest.java
Comment (pyaeger) [reply src/test/java/com/amplifier/controllers/ImgPostControllerIntTest.java]: Okay will get on that. Thank you!

File: src/test/java/com/amplifier/controllers/ImgPostControllerIntTest.java
Comment (pyaeger) [reply src/test/java/com/amplifier/controllers/ImgPostControllerIntTest.java]: Okay will do. Thx!

**Review by L3viathon — APPROVED** [review]

**Pull request merged**
